gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePublishPlugin.java

```diff
@@ -31,21 +33,25 @@ public void apply(Project project) {
         project.getPluginManager().apply(MavenPublishPlugin.class);
         project.getPluginManager().apply(ConjurePlugin.class);
 
-        CompileIrTask compileIr = (CompileIrTask) project.getTasks().findByName(ConjurePlugin.CONJURE_IR);
-        if (compileIr == null) {
-            throw new GradleException("Unable to find compileIr task");
+        TaskProvider<CompileIrTask> compileIr = null;
+        try {
+            compileIr = project.getTasks().named(ConjurePlugin.CONJURE_IR, CompileIrTask.class);
+        } catch (UnknownDomainObjectException e) {
```

Comment:
Nit: Would move this block into its own method so we don't need to do the `null` initialization and reassignment to a final variable.

Reply:
done!